addons/web/controllers/dataset.py

```diff
@@ -15,7 +16,10 @@
 
 def _call_kw_readonly(registry, request):
     params = request.get_json_data()['params']
-    model_class = registry[params['model']]
+    try:
+        model_class = registry[params['model']]
+    except KeyError as e:
+        raise BadRequest() from e
```

Comment:
Semantically speaking, `NotFound` is more appropriate here. Bonus point: it means we don't have to change JsonRpcDispatcher's `_handle_error`